### ISSUE TITLE
2.0 - verifyPlugin seems to ignore some specified `ide`s

### ISSUE THREAD

**Issue by PawelLipski** [JetBrains/intellij-platform-gradle-plugin#1637]
### What happened?

I'm running `verifyPlugin` task that's configured [in the following way](https://github.com/VirtusLab/git-machete-intellij-plugin/pull/1859/files):

```kotlin
  verifyPlugin {
    ides {
      ide(IntelliJPlatformType.IntellijIdeaCommunity, "2022.3")
      ide(IntelliJPlatformType.IntellijIdeaCommunity, "2023.1")
      ide(IntelliJPlatformType.IntellijIdeaCommunity, "2023.2")
      ide(IntelliJPlatformType.IntellijIdeaCommunity, "2023.3")
      ide(IntelliJPlatformType.IntellijIdeaCommunity, "2024.1")
    }
  }
```

But the verification &mdash; judging by the output &mdash; only happens for 2022.3 and 2024.1, but not the versions in-between.

 Also, `build/reports/pluginVerifier` only contains:

```
$ ls -1 build/reports/pluginVerifier
IC-223.7571.182
IC-241.14494.240
```

### Relevant log output or stack trace

<details>
<summary>Output of `./gradlew verifyPlugin`</summary>
<pre>
> Task :patchPluginXml
[org.jetbrains.intellij.platform] Patching plugin.xml: value of 'name[Git Machete]' tag will be set to 'git-machete'

> Task :verifyPluginProjectConfiguration
[org.jetbrains.intellij.platform] The following plugin configuration issues were found:
- The since-build='223' is lower than the target IntelliJ Platform major version: '242'.
- The Java configuration specifies sourceCompatibility='17' but IntelliJ Platform '2024.2' requires sourceCompatibility='21'.
- The Kotlin configuration specifies languageVersion='1.7' but IntelliJ Platform '2024.2' requires languageVersion='1.9'.

> Task :verifyPlugin
Starting the IntelliJ Plugin Verifier 1.365
Verification reports directory: /Users/pawel_lipski/git-machete-intellij-plugin/build/reports/pluginVerifier
2024-05-30T18:23:35 [main] INFO  verification - Reading IDE /Users/pawel_lipski/.cache/pluginVerifier/ides/IC-2022.3
2024-05-30T18:23:35 [main] INFO  c.j.p.options.OptionsParser - Reading IDE from /Users/pawel_lipski/.cache/pluginVerifier/ides/IC-2022.3
2024-05-30T18:23:35 [main] INFO  c.j.p.options.OptionsParser - Using Java runtime from /Users/pawel_lipski/.gradle/caches/transforms-4/df79472c8bb20377915bb624ef7cb6f4/transformed/ideaIC-242.10180.25/jbr/Contents/Home
2024-05-30T18:23:42 [main] INFO  verification - Reading IDE /Users/pawel_lipski/.cache/pluginVerifier/ides/IC-2024.1
2024-05-30T18:23:42 [main] INFO  c.j.p.options.OptionsParser - Reading IDE from /Users/pawel_lipski/.cache/pluginVerifier/ides/IC-2024.1
2024-05-30T18:23:42 [main] INFO  c.j.p.options.OptionsParser - Using Java runtime from /Users/pawel_lipski/.gradle/caches/transforms-4/df79472c8bb20377915bb624ef7cb6f4/transformed/ideaIC-242.10180.25/jbr/Contents/Home
2024-05-30T18:23:56 [main] INFO  verification - Reading plugin to check from /Users/pawel_lipski/git-machete-intellij-plugin/build/distributions/git-machete-intellij-plugin-5.0.0-12-SNAPSHOT+git.ab1581d-dirty.zip
2024-05-30T18:24:00 [main] INFO  verification - Task check-plugin parameters:
Scheduled verifications (2):
com.virtuslab.git-machete:5.0.0-12-SNAPSHOT+git.ab1581d-dirty against IC-223.7571.182, com.virtuslab.git-machete:5.0.0-12-SNAPSHOT+git.ab1581d-dirty against IC-241.14494.240

2024-05-30T18:24:04 [main] INFO  verification - Finished 1 of 2 verifications (in 3.6 s): IC-241.14494.240 against com.virtuslab.git-machete:5.0.0-12-SNAPSHOT+git.ab1581d-dirty: Compatible. 23 usages of deprecated API. 1 usage of experimental API
2024-05-30T18:24:04 [main] INFO  verification - Finished 2 of 2 verifications (in 3.7 s): IC-223.7571.182 against com.virtuslab.git-machete:5.0.0-12-SNAPSHOT+git.ab1581d-dirty: Compatible. 1 usage of deprecated API. 1 usage of experimental API
Plugin com.virtuslab.git-machete:5.0.0-12-SNAPSHOT+git.ab1581d-dirty against IC-241.14494.240: Compatible. 23 usages of deprecated API. 1 usage of experimental API
Deprecated API usages (23):
    #Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL access
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.file.highlighting.MacheteAnnotator.processMacheteGeneratedEntry(MacheteGeneratedEntry, AnnotationHolder) : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.actions.traverse.TraverseSyncToRemote.execute() : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.actions.traverse.CheckoutAndExecute.checkoutAndExecuteOnUIThread(GitRepository, BaseEnhancedGraphTable, String, Runnable) : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.ui.impl.table.EnhancedGraphTable.notifyAboutUnmanagedBranch(ILocalBranchReference, String) : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.ui.impl.RediscoverSuggester$1.run(ProgressIndicator) : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.ui.impl.errorreport.GitMacheteErrorReportSubmitter.submit(IdeaLoggingEvent[], String, Component, Consumer) : boolean
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.actions.toolbar.DiscoverAction.saveDiscoveredLayout(IGitMacheteRepositorySnapshot, Path, Project, IBranchLayoutWriter, Runnable) : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.ui.impl.backgroundables.GitMacheteRepositoryUpdateBackgroundable.run(ProgressIndicator) : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.file.ReparseMacheteFileOnGitRepositoryChange.repositoryChanged(GitRepository) : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.ui.impl.table.EnhancedGraphTable.queueRepositoryUpdateAndModelRefresh(Runnable) : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.ui.impl.table.EnhancedGraphTable.lambda$subscribeToGitRepositoryFilesChanges$1(GitRepository) : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.ui.impl.table.EnhancedGraphTable$3.onDiscoverFailure() : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.actions.common.SlideOut.lambda$run$1(Runnable) : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.actions.traverse.TraverseSyncToParent.execute() : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.ui.impl.table.EnhancedGraphTable$3.onDiscoverSuccess(IGitMacheteRepository, IGitMacheteRepositorySnapshot) : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.actions.common.SlideOut.lambda$handleBranchDeletionDecision$6(boolean, Runnable) : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.actions.common.SlideOut.lambda$run$3(Runnable, Boolean) : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.ui.impl.gitrepositoryselection.GitRepositoryComboBox.lambda$new$1() : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.actions.toolbar.DiscoverAction$1.doRun(ProgressIndicator) : void
        Deprecated field com.intellij.openapi.application.ModalityState.NON_MODAL : com.intellij.openapi.application.ModalityState is accessed in com.virtuslab.gitmachete.frontend.ui.impl.table.EnhancedGraphTablePopupMenuListener$1.run() : void
    #Deprecated method git4idea.branch.GitBrancher.merge(String, GitBrancher.DeleteOnMergeOption, List) invocation
        Deprecated method git4idea.branch.GitBrancher.merge(java.lang.String arg0, git4idea.branch.GitBrancher.DeleteOnMergeOption arg1, java.util.List arg2) : void is invoked in com.virtuslab.gitmachete.frontend.actions.base.BaseSyncToParentByMergeAction.lambda$doMergeIntoNonCurrentBranch$2(GitBrancher, String, List) : void
        Deprecated method git4idea.branch.GitBrancher.merge(java.lang.String arg0, git4idea.branch.GitBrancher.DeleteOnMergeOption arg1, java.util.List arg2) : void is invoked in com.virtuslab.gitmachete.frontend.actions.base.BaseSyncToParentByMergeAction.doMergeIntoCurrentBranch(GitRepository, MergeProps) : void
    #Deprecated method com.intellij.dvcs.DvcsUtil.guessCurrentRepositoryQuick(Project, AbstractRepositoryManager, String) invocation
        Deprecated method com.intellij.dvcs.DvcsUtil.guessCurrentRepositoryQuick(com.intellij.openapi.project.Project project, com.intellij.dvcs.repo.AbstractRepositoryManager manager, java.lang.String defaultRootPathValue) : T is invoked in com.virtuslab.gitmachete.frontend.actions.toolbar.OpenMacheteFileAction.actionPerformed(AnActionEvent) : void
Experimental API usages (1):
    #Experimental API interface com.intellij.vcs.log.ui.MainVcsLogUi reference
        Experimental API interface com.intellij.vcs.log.ui.MainVcsLogUi is referenced in com.virtuslab.gitmachete.frontend.actions.contextmenu.ShowSelectedInGitLogAction.lambda$actionPerformed$1(Project, Hash, MainVcsLogUi) : void. This interface can be changed in a future release leading to incompatibilities
Dynamic Plugin Eligibility:
    Plugin can probably be enabled or disabled without IDE restart

Plugin com.virtuslab.git-machete:5.0.0-12-SNAPSHOT+git.ab1581d-dirty against IC-223.7571.182: Compatible. 1 usage of deprecated API. 1 usage of experimental API
Deprecated API usages (1):
    #Deprecated method com.intellij.dvcs.DvcsUtil.guessCurrentRepositoryQuick(Project, AbstractRepositoryManager, String) invocation
        Deprecated method com.intellij.dvcs.DvcsUtil.guessCurrentRepositoryQuick(com.intellij.openapi.project.Project project, com.intellij.dvcs.repo.AbstractRepositoryManager manager, java.lang.String defaultRootPathValue) : T is invoked in com.virtuslab.gitmachete.frontend.actions.toolbar.OpenMacheteFileAction.actionPerformed(AnActionEvent) : void
Experimental API usages (1):
    #Experimental API interface com.intellij.vcs.log.ui.MainVcsLogUi reference
        Experimental API interface com.intellij.vcs.log.ui.MainVcsLogUi is referenced in com.virtuslab.gitmachete.frontend.actions.contextmenu.ShowSelectedInGitLogAction.lambda$actionPerformed$1(Project, Hash, MainVcsLogUi) : void. This interface can be changed in a future release leading to incompatibilities
Dynamic Plugin Eligibility:
    Plugin can probably be enabled or disabled without IDE restart

2024-05-30T18:24:04 [main] INFO  verification - Total time spent downloading plugins and their dependencies: 0 ms
2024-05-30T18:24:04 [main] INFO  verification - Total amount of plugins and dependencies downloaded: 0 B
2024-05-30T18:24:04 [main] INFO  verification - Total amount of space used for plugins and dependencies: 0 B
2024-05-30T18:24:04 [main] INFO  verification - Verification reports for com.virtuslab.git-machete:5.0.0-12-SNAPSHOT+git.ab1581d-dirty saved to /Users/pawel_lipski/git-machete-intellij-plugin/build/reports/pluginVerifier/IC-241.14494.240
2024-05-30T18:24:04 [main] INFO  verification - Verification reports for com.virtuslab.git-machete:5.0.0-12-SNAPSHOT+git.ab1581d-dirty saved to /Users/pawel_lipski/git-machete-intellij-plugin/build/reports/pluginVerifier/IC-223.7571.182
2024-05-30T18:24:04 [main] INFO  verification - Total time spent in plugin verification: 28 s 679 ms

BUILD SUCCESSFUL in 2m 54s
147 actionable tasks: 15 executed, 132 up-to-date
</pre>
</details>

### Steps to reproduce

Check out https://github.com/VirtusLab/git-machete-intellij-plugin/pull/1859 and run `./gradlew verifyPlugin`.

### Gradle IntelliJ Plugin version

2.0.0-beta5

### Gradle version

8.7

### Operating System

macOS

### Link to build, i.e. failing GitHub Action job

https://app.circleci.com/pipelines/github/VirtusLab/git-machete-intellij-plugin/10104/workflows/f03cebca-71fc-4a0c-bd0d-10d0b47d4cb8/jobs/10547

**Comment by hsz** [comment]
Thank you for reporting. Fix is now available in `2.0.0-beta6`

**Comment by PawelLipski** [comment]
Whoops should I change my config in any specific way? I've just retried `verifyPlugin`... and all versions other than the last specified `ide` are ignored ☹️ [Link to CI logs](https://app.circleci.com/pipelines/github/VirtusLab/git-machete-intellij-plugin/10125/workflows/7e8c9ee6-9dcb-4831-a96f-4b33f8f0995b/jobs/10568/parallel-runs/0/steps/0-114), same happening locally
